### PR TITLE
fix (cache): support for DB debug logging

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

**Review by jefferai — APPROVED** [review]

File: internal/clientcache/internal/daemon/server.go [hunk 1]
```diff
@@ -560,3 +568,17 @@ const (
 	dbFileName = "cache.db"
 	fkPragma   = "?_pragma=foreign_keys(1)"
 )
+
+type gormDebugLogger struct {
+	hclog.Logger
+}
+
+func (g gormDebugLogger) Printf(msg string, values ...interface{}) {
```
Comment (johanbrandhorst) [inline internal/clientcache/internal/daemon/server.go:576]:
```suggestion
func (g gormDebugLogger) Printf(msg string, values ...any) {
```

File: internal/clientcache/internal/db/db.go [hunk 2]
```diff
@@ -54,8 +54,11 @@ func Open(ctx context.Context, opt ...Option) (*db.DB, error) {
 	if err != nil {
 		return nil, errors.Wrap(ctx, err, op)
 	}
-	conn.Debug(opts.withDebug)
-
+	defer func() {
+		// let's not capture the output of resetSchema and createTables, so
+		// we'll defer turning on debug
+		conn.Debug(opts.withDebug)
+	}()
```
Comment (johanbrandhorst) [inline internal/clientcache/internal/db/db.go:61]: Is this that noisy? This seems useful to leave enabled, and I could see it confusing some future dev that they enabled debug log but can't see the initial queries.

Comment (jimlambrt) [reply internal/clientcache/internal/db/db.go:61]: It just doesn't seem to provide much value to see a wall of create this, that and the other db obj.  Typically when you're debugging, you're looking for the queries I think.

Comment (jimlambrt) [reply internal/clientcache/internal/db/db.go:61]: In the future, it's pretty simple to discover where `createTable(...)` is called and to see the comment I left.

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**